sympy/core/mul.py

```diff
@@ -65,7 +65,9 @@ def _unevaluated_Mul(*args):
     False
 
     """
-    args = list(args)
+    sifted = sift(args, lambda x: bool(x.is_commutative))
+    args = sifted[True]
+    nc = sifted[False]
```

Comment:
`nc` can be potentially buggy variable name to use because it is assigned inside loop 
`c, nc = a.args_cnc()`